drivers/mbox/mbox_ti_omap.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2024 Texas Instruments Incorporated.
+ *
+ * TI OMAP Mailbox driver for Zephyr's MBOX model.
+ */
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/mbox.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/irq.h>
+
+#define LOG_LEVEL CONFIG_MBOX_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ti_omap_mailbox);
+
+#define DT_DRV_COMPAT ti_omap_mailbox
+
+#define MAILBOX_IRQ_NEWMSG(m)		(1 << (2 * (m)))
+#define MAILBOX_IRQ_NOTFULL(m)		(1 << (2 * (m) + 1))
+#define OMAP_MAILBOX_NUM_MSGS 16
+#define MAILBOX_MAX_CHANNELS 16
+#define OMAP_MAILBOX_NUM_USERS 4
+#define MAILBOX_MBOX_SIZE    sizeof(uint32_t)
+
+struct omap_mailbox_data {
+	mbox_callback_t cb[MAILBOX_MAX_CHANNELS];
+	void *user_data[MAILBOX_MAX_CHANNELS];
+	bool channel_enable[MAILBOX_MAX_CHANNELS];
+	uint32_t received_data;
+	void *mapped_base;
+};
+
+struct omap_mailbox_config {
+	uint32_t usr_id;
+};
+
+struct omap_mailbox_irq_regs {
+	uint32_t	status_raw;
+	uint32_t	status_clear;
+	uint32_t	enable_set;
+	uint32_t	enable_clear;
+};
+
+struct omap_mailbox_regs {
+	uint32_t	revision;
+	uint32_t	__pad0[3];
+	uint32_t	sysconfig;
+	uint32_t	__pad1[11];
+	uint32_t	message[OMAP_MAILBOX_NUM_MSGS];
+	uint32_t	fifo_status[OMAP_MAILBOX_NUM_MSGS];
+	uint32_t	msg_status[OMAP_MAILBOX_NUM_MSGS];
+	struct omap_mailbox_irq_regs irq_regs[OMAP_MAILBOX_NUM_USERS];
+};
+
+static void omap_mailbox_isr(const struct device *dev)
+{
+	struct omap_mailbox_data *data = dev->data;
+	const struct omap_mailbox_config *cfg = dev->config;
+
+	volatile struct omap_mailbox_regs *regs = (struct omap_mailbox_regs *)data->mapped_base;
```

Comment:
You could consider using defines to access driver related data like in https://github.com/zephyrproject-rtos/zephyr/blob/9f0cd47c784e4d2099f59899d17a6ba5f3d4be55/drivers/gpio/gpio_davinci.c#L26